test/starknet/vanilla_space.ts

```diff
@@ -48,31 +49,59 @@ async function setup() {
   };
 }
 
+function setup_calldata(
+  proposer_address: bigint,
+  execution_hash: bigint,
+  metadata_uri: Array<bigint>,
+  eth_block_number: bigint,
+  params: Array<bigint>
+): Array<bigint> {
+  const calldata = [proposer_address, execution_hash];
+  calldata.push(BigInt(metadata_uri.length));
+
+  for (let i = 0; i < metadata_uri.length; i++) {
+    calldata.push(metadata_uri[i]);
+  }
+
+  calldata.push(eth_block_number);
+  calldata.push(BigInt(params.length));
+
+  for (let i = 0; i < params.length; i++) {
+    calldata.push(params[i]);
+  }
+
+  return calldata;
```

Comment:
Try this, that's much more readable
```suggestion
  return [
    proposer_address,
    execution_hash,
    BigInt(metadata_uri.length),
    ...metadata_uri,
    eth_block_number,
    BigInt(params.length),
    ...params
  ];
```

Reply:
(You may not need a function in the end)